### ISSUE TITLE
Maven version 0.4.1 is missing dependencies

### ISSUE THREAD

**Issue by ShadelessFox** [ClickHouse/clickhouse-java#1257]
### Describe the bug

Using the driver is impossible as simple as just adding a Maven dependency because its `pom.xml` lacks all dependencies whatsoever:

https://repo1.maven.org/maven2/com/clickhouse/clickhouse-jdbc/0.4.1/clickhouse-jdbc-0.4.1.pom

In contrast, the previous version doesn't have the same flaw:

https://repo1.maven.org/maven2/com/clickhouse/clickhouse-jdbc/0.4.0/clickhouse-jdbc-0.4.0.pom

It's also problematic to add the required dependencies manually because they aren't listed anywhere. Perhaps I'm missing something?

**Comment by zhicwu** [comment]
Thanks for the report @ShadelessFox. This was introduced in #1204 for making all dependencies optional. However, I completely forgot to replace the default jar with a shaded one 😞 This will be fixed in next release.

By the way, why DBeaver does not use shaded jar by specifying classifier?

**Comment by ShadelessFox** [comment]
Are you talking about artifact classifiers? Can you elaborate, please?

We can specify a direct link to the jar file, but we'd better wait for the next release.

**Comment by zhicwu** [comment]
Yes, it's better to wait for the next release. RE: artifact classifier, I was talking about something like the following:
```xml
<dependency>
    <groupId>com.clickhouse</groupId>
    <artifactId>clickhouse-jdbc</artifactId>
    <version>0.4.1</version>
    <classifier>all</classifier>
</dependency>
```

**Comment by ShadelessFox** [comment]
Ah, I see. Yes, specifying the classifier works, indeed. We can go with it for now.

Can this ticket remain open until the new version is released so we can update our dependencies?

**Comment by zhicwu** [comment]
Thanks but I think you don't have to make the change in DBeaver but wait for the new release.

Sure, let's leave this open until the new version is released.

**Comment by zhicwu** [comment]
@ShadelessFox, I added a few more dependencies with `provided` scope, so that they will show up in DBeaver when downloading driver files. I'm not sure how I can test using DBeaver, could you help to validate using nightly build? It should work without adding classifier.

```xml
...
<repositories>
    <repository>
        <id>ossrh</id>
        <name>Sonatype OSSRH</name>
        <url>https://s01.oss.sonatype.org/content/repositories/snapshots/</url>
    </repository>
</repositories>
...
<dependency>
    <groupId>com.clickhouse</groupId>
    <artifactId>clickhouse-jdbc</artifactId>
    <version>0.4.2-SNAPSHOT</version>
</dependency>
```

**Comment by ShadelessFox** [comment]
Thank you for providing a nightly version, but we don't want to add a new repository just for that.

We decided to specify the classifier until a new version is released.

**Comment by zhicwu** [comment]
> Thank you for providing a nightly version, but we don't want to add a new repository just for that.

No, I was asking if you can validate the change using nightly build at your end as I don't know how :)

**Comment by ShadelessFox** [comment]
So just locally? No, it doesn't work against the provided repository.

DBeaver expects to find file `clickhouse-jdbc-0.4.2-SNAPSHOT.pom` under `com/clickhouse/clickhouse-java/0.4.2-SNAPSHOT`, but it's missing. Looks like all required information is contained within `maven-metadata.xml` which we don't handle.

**Comment by zhicwu** [comment]
Just released v0.4.2 and tried on DBeaver, it looks good except classifier can be removed to reduce download size.

![image](https://user-images.githubusercontent.com/4270380/226641587-59c9b3a7-245e-4a4e-b72a-ef0003717c0d.png)
